### ISSUE TITLE
Maximum retries of <> reached without finding a unique value 

### ISSUE THREAD

**Issue by dixitk13** [muonsoft/openapi-mock#30]
So I have a swagger-mock running inside docker on an `openapi: 3.0.0` yaml file.

I keep getting `OverflowException: Maximum retries of 10000 reached without finding a unique value in /app/vendor/fzaninotto/faker/src/Faker/UniqueGenerator.php:51`

I can't share the YAML file but would like to know if there's something which I might be doing wrong.

**Comment by strider2038** [comment]
Hello! It is hard to say without a specification file. This is definitely a problem with uniqueness of values. Uniqueness problem can occur with a big array with `uniqueItems: true` and a few possible variants of values. For example, something like this will always lead to such errors.

```yaml
type: array
uniqueItems: true
minItems: 3
items:
    type: string
    enum:
        - one
        - two
```

Maybe `docker logs swagger_mock` can help with more details.

**Comment by dixitk13** [comment]
this is all I get:

```
OverflowException: Maximum retries of 10000 reached without finding a unique value in /app/vendor/fzaninotto/faker/src/Faker/UniqueGenerator.php:51
Stack trace:
#0 /app/src/Mock/Generation/Value/Composite/HashMapValueGenerator.php(79): Faker\UniqueGenerator->__call('word', Array)
#1 /app/src/Mock/Generation/Value/Composite/HashMapValueGenerator.php(56): App\Mock\Generation\Value\Composite\HashMapValueGenerator->generateAndAppendRandomProperties(Object(App\Mock\Parameters\Schema\Type\Composite\HashMapType), Object(stdClass))
#2 /app/src/Mock/Generation/Value/Composite/HashMapValueGenerator.php(46): App\Mock\Generation\Value\Composite\HashMapValueGenerator->generateHashMap(Object(App\Mock\Parameters\Schema\Type\Composite\HashMapType))
#3 /app/src/Mock/Generation/Value/Composite/ObjectValueGenerator.php(63): App\Mock\Generation\Value\Composite\HashMapValueGenerator->generateValue(Object(App\Mock\Parameters\Schema\Type\Composite\HashMapType))
#4 /app/src/Mock/Generation/Value/Composite/ObjectValueGenerator.php(52): App\Mock\Generation\Value\Composite\ObjectValueGenerator->generateValueByType(Object(App\Mock\Parameters\Schema\Type\Composite\HashMapType))
#5 /app/src/Mock/Generation/Value/Composite/ObjectValueGenerator.php(36): App\Mock\Generation\Value\Composite\ObjectValueGenerator->generateObject(Object(App\Mock\Parameters\Schema\Type\Composite\ObjectType))
#6 /app/src/Mock/Generation/Value/Composite/ArrayGenerator/ArrayRegularValueGenerator.php(37): App\Mock\Generation\Value\Composite\ObjectValueGenerator->generateValue(Object(App\Mock\Parameters\Schema\Type\Composite\ObjectType))
#7 /app/src/Mock/Generation/Value/Composite/ArrayGenerator/DelegatingArrayValueGenerator.php(60): App\Mock\Generation\Value\Composite\ArrayGenerator\ArrayRegularValueGenerator->generateArray(Object(App\Mock\Generation\Value\Composite\ObjectValueGenerator), Object(App\Mock\Parameters\Schema\Type\Composite\ArrayType))
#8 /app/src/Mock/Generation/Value/Composite/ArrayGenerator/DelegatingArrayValueGenerator.php(47): App\Mock\Generation\Value\Composite\ArrayGenerator\DelegatingArrayValueGenerator->generateArray(Object(App\Mock\Parameters\Schema\Type\Composite\ArrayType))
#9 /app/src/Mock/Generation/DataGenerator.php(36): App\Mock\Generation\Value\Composite\ArrayGenerator\DelegatingArrayValueGenerator->generateValue(Object(App\Mock\Parameters\Schema\Type\Composite\ArrayType))
#10 /app/src/Mock/MockResponseGenerator.php(62): App\Mock\Generation\DataGenerator->generateData(Object(App\Mock\Parameters\Schema\Schema))
#11 /app/src/API/RequestHandler.php(43): App\Mock\MockResponseGenerator->generateResponse(Object(Symfony\Component\HttpFoundation\Request), Object(App\Mock\Parameters\Endpoint))
#12 /app/src/EventListener/RequestListener.php(32): App\API\RequestHandler->handleRequest(Object(Symfony\Component\HttpFoundation\Request))
#13 /app/vendor/symfony/event-dispatcher/EventDispatcher.php(298): App\EventListener\RequestListener->onKernelRequest(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request', Object(Symfony\Component\EventDispatcher\EventDispatcher))
#14 /app/vendor/symfony/event-dispatcher/EventDispatcher.php(260): Symfony\Component\EventDispatcher\EventDispatcher::Symfony\Component\EventDispatcher\{closure}(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request', Object(Symfony\Component\EventDispatcher\EventDispatcher))
#15 /app/vendor/symfony/event-dispatcher/EventDispatcher.php(235): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'kernel.request', Object(Symfony\Component\HttpKernel\Event\RequestEvent))
#16 /app/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners(Array, 'kernel.request', Object(Symfony\Component\HttpKernel\Event\RequestEvent))
#17 /app/vendor/symfony/http-kernel/HttpKernel.php(127): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Symfony\Component\HttpKernel\Event\RequestEvent), 'kernel.request')
#18 /app/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#19 /app/vendor/symfony/http-kernel/Kernel.php(198): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#20 /app/bin/worker.php(48): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#21 {main}
```

and I've set the config like so : 
```
      SWAGGER_MOCK_LOG_LEVEL: debug 
```

docker-logs
```
11:34 $ docker-compose logs -f
Attaching to swagger_mock
swagger_mock    | 
swagger_mock    |  // Clearing the cache for the prod environment with debug                      
swagger_mock    |  // false                                                                       
swagger_mock    | 
swagger_mock    |  [OK] Cache for the "prod" environment (debug=false) was successfully cleared.  
swagger_mock    | 
swagger_mock    | 
swagger_mock    |  // Warming up the cache for the prod environment with debug                    
swagger_mock    |  // false                                                                       
swagger_mock    | 
swagger_mock    |  [OK] Cache for the "prod" environment (debug=false) was successfully warmed.   
swagger_mock    | 


swagger_mock    | INFO[0009] 172.18.0.1 {248ms} 500 GET http://localhost:8080/<endpoint>



swagger_mock    | INFO[0015] 172.18.0.1 {203ms} 500 GET http://localhost:8080/<endpoint>
swagger_mock    | INFO[0016] 172.18.0.1 {193ms} 500 GET http://localhost:8080/<endpoint> 
```

and to add to the puzzle, i do get data sometimes, like once in 20/30 attempts irregularly

**Comment by dixitk13** [comment]
It is worth noting, it works for certain APIs flawlessly but not every.

**Comment by strider2038** [comment]
According to logs, the problem is with hash map type <https://swagger.io/docs/specification/data-models/dictionaries/>. Maybe you have something like this in the specification?

```yaml
type: object
additionalProperties:
  type: object
  properties:
    value:
      # it can be limited strings or enums
      type: string
      maxLength: 3
```

Can you make specification reproducing this bug? It is really hard to guess what can be wrong.

**Comment by strider2038** [comment]
This problem was solved in version [v0.2](https://github.com/muonsoft/openapi-mock/releases/tag/v0.2.0). Be aware, the project was moved to another location. Please, read [project description](https://github.com/muonsoft/openapi-mock/blob/master/README.md) for more information.
